internal/jet/job.go

```diff
@@ -0,0 +1,143 @@
+package jet
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/hazelcast/hazelcast-go-client"
+	"github.com/hazelcast/hazelcast-go-client/types"
+
+	"github.com/hazelcast/hazelcast-commandline-client/internal/cluster"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/log"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/proto/codec"
+	"github.com/hazelcast/hazelcast-commandline-client/internal/proto/codec/control"
+)
+
+type spinner interface {
```

Comment:
Do you remember that you suggested to pass an updater function while I was implementing custom class download/upload (`progressSetter`). Do we really need this extra interface as we already have a spinner interface inside `clc/common.go`?

Reply:
Yes. The primary motivation for my comment was decoupling code in the API from the commands. Doing that via n interface may be better, since later we may have to add more methods to this interface.